src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -58,7 +58,7 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
 
     if (!program.compilerOptions.noEmit && !program.hasError()) {
         // Write out the dotnet model to the output path
-        const root = createModel(context);
+        const [root, azureArm] = createModel(context);
         if (
```

Comment:
From your description, we could pass this flag by configuration, why need to be part of the created model?

Reply:
It is from `SdkContext.arm` in tcgc, which is invoked during `createModel`

Reply:
Mingzhe is working on singleton sdkContext. After that, you could read the sdkContext.arm without changing the return type.

Reply:
That would be great, @archerzz any ETA for the singleton sdkContext to be available?

Reply:
This change is not needed anymore

Reply:
> That would be great, @archerzz any ETA for the singleton sdkContext to be available?

@live1206 It's in https://github.com/Azure/autorest.csharp/pull/4091

Reply:
@pshao25 
Right now, we have duplicated `createSdkContext` in `clientModelBuilder.ts` and `operation.ts`.
In #4091, we are consolidating to have only 1 in `clientModelBuilder.ts` and pass along it to `operation.ts`.
In this PR, I moved `createSdkContext` to `emitter.ts` and pass along it to `clientModelBuilder.ts`.
In the end, there is still a single `createSdkContext` call, no dependency from each other.

Reply:
The latest code looks good. Just one more comment: sdkContext.emitContext = context, so createModel only need one parameter sdkContext.

Reply:
updated